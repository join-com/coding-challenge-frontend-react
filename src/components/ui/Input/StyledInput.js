import styled from "styled-components";
import { defaultInputStyles, hoverBorderStyles } from "../../../helpers/stylesMixins";

export const StyledInput = styled.input`
  ${defaultInputStyles};
  width: 100%;
  box-sizing: border-box;
  &:focus {
    ${({ disabled }) => !disabled && hoverBorderStyles};
  }
`;
