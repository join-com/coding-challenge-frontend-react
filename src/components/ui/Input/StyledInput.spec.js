import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { StyledInput } from "./StyledInput";

describe("Styled StyledInput component", () => {
  const defaultProps = {
    value: "",
    placeholder: "",
    disabled: false,
    uppercase: false,
    type: "text",
    size: "normal",
    name: "name",
    id: "id",
    onChange: jest.fn()
  };

  const getWrapper = (testProps = {}) => {
    const props = { ...defaultProps, ...testProps };
    return renderer.create(<StyledInput {...props} />).toJSON();
  };

  it("should match snapshot ", () => {
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });
  it("should match snapshot for disabled prop ", () => {
    const wrapper = getWrapper({ disabled: true });
    expect(wrapper).toMatchSnapshot();
  });
});
