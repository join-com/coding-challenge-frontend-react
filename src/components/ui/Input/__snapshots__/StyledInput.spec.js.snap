// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled StyledInput component should match snapshot  1`] = `
.c0 {
  border: solid 2px #1D232B;
  border-radius: 3px;
  font-family: 'Oswald',sans-serif;
  font-size: 1rem;
  font-weight: bold;
  line-height: 1.2;
  background-color: #F9F9F9;
  color: #1D232B;
  opacity: 1;
  padding: 0.5rem;
  outline: none;
  -webkit-transition: all 0.1s ease-in;
  transition: all 0.1s ease-in;
  border-radius: 3px;
  text-transform: none;
  cursor: pointer;
  width: 100%;
  box-sizing: border-box;
}

.c0:hover {
  border-color: #90D3DC;
}

.c0:focus {
  border-color: #90D3DC;
}

<input
  className="c0"
  disabled={false}
  id="id"
  name="name"
  onChange={[MockFunction]}
  placeholder=""
  size="normal"
  type="text"
  value=""
/>
`;

exports[`Styled StyledInput component should match snapshot for disabled prop  1`] = `
.c0 {
  border: solid 2px #1D232B;
  border-radius: 3px;
  font-family: 'Oswald',sans-serif;
  font-size: 1rem;
  font-weight: bold;
  line-height: 1.2;
  background-color: #F9F9F9;
  color: #1D232B;
  opacity: 0.5;
  padding: 0.5rem;
  outline: none;
  -webkit-transition: all 0.1s ease-in;
  transition: all 0.1s ease-in;
  border-radius: 3px;
  text-transform: none;
  cursor: not-allowed;
  width: 100%;
  box-sizing: border-box;
}

<input
  className="c0"
  disabled={true}
  id="id"
  name="name"
  onChange={[MockFunction]}
  placeholder=""
  size="normal"
  type="text"
  value=""
/>
`;
